monad-rpc/src/chainstate/mod.rs

```diff
@@ -59,9 +59,18 @@ pub struct ChainState<T> {
 #[derive(Debug)]
 pub enum ChainStateError {
     Triedb(String),
+    Archive(String),
     ResourceNotFound,
 }
 
+impl From<monad_archive::prelude::Report> for ChainStateError {
+    fn from(e: monad_archive::prelude::Report) -> Self {
+        // Log with debug to get error chain, but return only top level error in response
+        error!("Archive Error: {e:?}");
+        ChainStateError::Archive(e.to_string())
```

Comment:
Is there any concern about unsanitized errors from the archiver ? (i.e. end user seeing url of aws or mongo instances)

Reply:
I added test assertions that all `get` methods on ArchiveReader do not contain any details of the underlying store when formatted with Display. Full error chain is present on Debug which is good to log internally.